includes/forum.php

```diff
@@ -1842,56 +1855,67 @@ public function show_topic_menu($show_all_buttons = true) {
         return $menu;
     }
 
+    public function get_edit_history_link($post_id) {
+        return home_url('/forum/edithistory/' . $post_id);
+    }
+    
     // Generate post menu.
     public function show_post_menu($post_id, $author_id, $counter, $post_date) {
         $menu = '';
-
+    
         // Only show post-menu when the topic is approved.
         if ($this->approval->is_topic_approved($this->current_topic)) {
             if (is_user_logged_in()) {
                 $current_user_id = get_current_user_id();
-
+    
                 if ($this->permissions->can_delete_post($current_user_id, $post_id, $author_id, $post_date) && ($counter > 1 || $this->current_page >= 1)) {
                     // Delete button.
                     $delete_post_link = $this->get_link('topic', $this->current_topic, array(
-						'post'        => $post_id,
-						'remove_post' => 1,
-						'_wpnonce'    => wp_create_nonce('asgaros_forum_delete_post'),
-					));
-
+                        'post'        => $post_id,
+                        'remove_post' => 1,
+                        '_wpnonce'    => wp_create_nonce('asgaros_forum_delete_post'),
+                    ));
+    
                     $menu     .= '<a class="delete-forum-post" onclick="return confirm(\''.__('Are you sure you want to remove this?', 'asgaros-forum').'\');" href="'.$delete_post_link.'">';
                         $menu .= '<span class="menu-icon fas fa-trash-alt"></span>';
                         $menu .= __('Delete', 'asgaros-forum');
                     $menu     .= '</a>';
                 }
-
+    
                 if ($this->permissions->can_edit_post($current_user_id, $post_id, $author_id, $post_date)) {
                     // Edit button.
                     $menu     .= '<a href="'.$this->get_link('editpost', $post_id, array('part' => ($this->current_page + 1))).'">';
                         $menu .= '<span class="menu-icon fas fa-pencil-alt"></span>';
                         $menu .= __('Edit', 'asgaros-forum');
                     $menu     .= '</a>';
-                }
-            }
-
-            if ($this->permissions->isModerator('current') || (!$this->is_topic_closed($this->current_topic) && ((is_user_logged_in() && !$this->permissions->isBanned('current')) || (!is_user_logged_in() && $this->options['allow_guest_postings'])))) {
-                // Quote button.
-                $menu     .= '<a class="forum-editor-quote-button" data-value-id="'.$post_id.'" href="'.$this->get_link('addpost', $this->current_topic, array('quote' => $post_id)).'">';
-                    $menu .= '<span class="menu-icon fas fa-quote-left"></span>';
-                    $menu .= __('Quote', 'asgaros-forum');
-                $menu     .= '</a>';
+    
+                    // Edit History link.
+                $edit_history_link = $this->get_edit_history_link($post_id);
+                $menu     .= '<a href="'.$edit_history_link.'">';
```

Comment:
```suggestion
                $menu     .= '<a href="'.htmlspecialchars($edit_history_link).'">';
```
- for example if the url has an `?foo&bar`, it should be encoded to `<a href="?foo&amp;bar` to remain valid HTML